src/main/java/edu/ucsb/courses/config/SecurityConfig.java

```diff
@@ -16,8 +16,13 @@
 
   @Override
   public void configure(HttpSecurity http) throws Exception {
-    http.authorizeRequests().mvcMatchers("/api/public/**").permitAll().mvcMatchers("/api/**")
-        .authenticated().anyRequest().permitAll();
+    http.authorizeRequests()
+    .mvcMatchers("/**").permitAll()
+    .mvcMatchers("/api/public/**").permitAll()
+    .mvcMatchers("/swagger-ui/**").permitAll()
+    .mvcMatchers("/api/docs").permitAll()
+    .mvcMatchers("/api/**")
+    .authenticated().anyRequest().permitAll();
```

Comment:
Would the paths `/api/public/`, /`swagger-ui`, etc not fall under `/**`? If that's the case, would we only need that first `permitAll` line?

Reply:
The paths `api/public` and `api/docs` are also covered by `api/**` so they definitely have to be called out separately.

Arguably the path `/swagger-ui/**` might be covered by the `**`.   However, I would suggest the redundancy is helpful for documentation purposes; someone modifying this code later may not be aware of that path, especially since a search of the project for paths will not reveal it anywhere in the code.

I've tried rearranging these lines to try to make the code more clear and expressive, and each time something broke.  So I'd like to suggest that we go with this, and then if you'd like to try submitting a PR to refactor it, give it a try.

Reply:
makes sense 👍